ibis/backends/tests/test_client.py

```diff
@@ -202,3 +205,245 @@ def test_separate_database(con, alternate_current_database, current_data_db):
+)
+def test_insert_no_overwite_from_table(con):
+    sch = ibis.schema(
+        [
+            ('first_name', 'string'),
+            ('last_name', 'string'),
+            ('department_name', 'string'),
+            ('salary', 'float64'),
+        ]
+    )
+
+    temp_table = 'temp_to_table'
+    temporary = _create_temp_table_with_schema(con, temp_table, sch)
+
+    df2 = pd.DataFrame(
+        {
+            'first_name': ['X', 'Y', 'Z'],
+            'last_name': ['A', 'B', 'C'],
+            'department_name': ['XX', 'YY', 'ZZ'],
+            'salary': [400.0, 500.0, 600.0],
+        }
+    )
+
+    from_table_name = 'temp_from_table'
+    from_table = _create_temp_table_with_schema(
+        con, from_table_name, sch, data=df2
+    )
+
+    con.insert(temp_table, obj=from_table_name, overwrite=False)
+    assert len(temporary.execute()) == 3
+    tm.assert_frame_equal(temporary.execute(), from_table.execute())
+
+
+@pytest.mark.only_on_backends(
+    SQLALCHEMY_BACKENDS, reason="run only if backend is SQLAlchemy based",
+)
+def test_insert_overwrite_from_table(con):
+    sch = ibis.schema(
+        [
+            ('first_name', 'string'),
+            ('last_name', 'string'),
+            ('department_name', 'string'),
+            ('salary', 'float64'),
+        ]
+    )
+
+    df = pd.DataFrame(
+        {
+            'first_name': ['A', 'B', 'C'],
+            'last_name': ['D', 'E', 'F'],
+            'department_name': ['AA', 'BB', 'CC'],
+            'salary': [100.0, 200.0, 300.0],
+        }
+    )
+
+    temp_table = 'temp_to_table'
+    temporary = _create_temp_table_with_schema(con, temp_table, sch, data=df)
+
+    df2 = pd.DataFrame(
+        {
+            'first_name': ['X', 'Y', 'Z'],
+            'last_name': ['A', 'B', 'C'],
+            'department_name': ['XX', 'YY', 'ZZ'],
+            'salary': [400.0, 500.0, 600.0],
+        }
+    )
+
+    from_table_name = 'temp_from_table'
+    from_table = _create_temp_table_with_schema(
+        con, from_table_name, sch, data=df2
+    )
+
+    con.insert(temp_table, obj=from_table_name, overwrite=True)
+    assert len(temporary.execute()) == 3
+    tm.assert_frame_equal(temporary.execute(), from_table.execute())
+
+
+@pytest.mark.only_on_backends(
+    SQLALCHEMY_BACKENDS, reason="run only if backend is SQLAlchemy based",
+)
+def test_insert_no_overwite_from_expr(con):
+    sch = ibis.schema(
+        [
+            ('first_name', 'string'),
+            ('last_name', 'string'),
+            ('department_name', 'string'),
+            ('salary', 'float64'),
+        ]
+    )
+
+    temp_table = 'temp_to_table'
+    temporary = _create_temp_table_with_schema(con, temp_table, sch)
+
+    df2 = pd.DataFrame(
+        {
+            'first_name': ['X', 'Y', 'Z'],
+            'last_name': ['A', 'B', 'C'],
+            'department_name': ['XX', 'YY', 'ZZ'],
+            'salary': [400.0, 500.0, 600.0],
+        }
+    )
```

Comment:
There is a lot of repetition in these tests. Like this dataframe or similar being repeated lots of times. Can you define a fixture or a constant, and try to use always the same data? If you need different data (but similar) you can make a copy of the dataframe, and change whatever you need).

Same for the schema. And anything that keeps repeating among tests. Write reusable code, and don't copy-paste anything, which is a very poor programming practice.

Reply:
Sure, will work on it!